### PR TITLE
Vanilla pipeline is failing intermittently while getting the Targetcl…

### PR DESCRIPTION
**Pull request by jkagliwal-px**
Added a polling logic to check the cluster health check and error-out if not in "Healthy" state. 

Supporting Logs - 
1. When cluster is HEALTHY - https://aetos.pwx.purestorage.com/resultSet/testSetID/281979
2. When cluster is NOT HEALTHY - Attached screenshot in the JIRA Ticket. 
     https://portworx.atlassian.net/browse/PA-1440

### CODE REVIEW

File: drivers/pds/targetcluster/targetcluster.go [hunk 1]
```diff
@@ -85,9 +85,13 @@ func (tc *TargetCluster) GetDeploymentTargetID(clusterID, tenantID string) (stri
 			targetClusterStatus = targetClusters[i].GetStatus()
 		}
 	}
-	if targetClusterStatus != "healthy" {
-		return "", fmt.Errorf("target Cluster is not in healthy state due to error : %v", err)
-	}
+	err = wait.Poll(DefaultRetryInterval, 5*time.Minute, func() (bool, error) {
+		if targetClusterStatus != "healthy" {
```
Comment (Madanagopal19) [inline drivers/pds/targetcluster/targetcluster.go:89]: targetClusterStatus will not get updated, unless you call 
`targetClusters, err := components.DeploymentTarget.ListDeploymentTargetsBelongsToTenant(tenantID)` this inside the polling method.

**Review by Madanagopal19 — CHANGES_REQUESTED** [review]

File: drivers/pds/targetcluster/targetcluster.go [hunk 2]
```diff
@@ -69,25 +69,29 @@ type TargetCluster struct {
...
-		if targetClusters[i].GetClusterId() == clusterID {
-			deploymentTargetID = targetClusters[i].GetId()
-			log.Infof("deploymentTargetID %v", deploymentTargetID)
-			log.InfoD("Cluster ID: %v, Name: %v,Status: %v", targetClusters[i].GetClusterId(), targetClusters[i].GetName(), targetClusters[i].GetStatus())
-			targetClusterStatus = targetClusters[i].GetStatus()
+	err = wait.Poll(DefaultRetryInterval, MaxTimeout, func() (bool, error) {
+		targetClusters, err := components.DeploymentTarget.ListDeploymentTargetsBelongsToTenant(tenantID)
+		var targetClusterStatus string
+		if err != nil {
+			return true, fmt.Errorf("error while listing deployments: %v", err)
```
Comment (Madanagopal19) [inline drivers/pds/targetcluster/targetcluster.go:76]: return false

File: drivers/pds/targetcluster/targetcluster.go [hunk 3]
```diff
@@ -69,25 +69,29 @@ type TargetCluster struct {
...
 		}
-	}
-	if targetClusterStatus != "healthy" {
-		return "", fmt.Errorf("target Cluster is not in healthy state due to error : %v", err)
-	}
+		if targetClusters == nil {
+			return true, fmt.Errorf("target cluster passed is not available to the account/tenant %v", err)
```
Comment (Madanagopal19) [inline drivers/pds/targetcluster/targetcluster.go:79]: return false, why are you returning true ? you need to return true once the condition is satisfied.

File: drivers/pds/targetcluster/targetcluster.go (hunk 1 above)
Comment (Madanagopal19) [reply drivers/pds/targetcluster/targetcluster.go:89]:
```
if targetClusterStatus == "healthy" {
   return true, nil
}
```

File: drivers/pds/targetcluster/targetcluster.go (hunk 3 above)
Comment (Madanagopal19) [reply drivers/pds/targetcluster/targetcluster.go:79]: and also error will be nil here, no need to pass err

**Review by Madanagopal19 — CHANGES_REQUESTED** [review]

File: drivers/pds/targetcluster/targetcluster.go [hunk 4]
```diff
@@ -69,24 +69,31 @@ type TargetCluster struct {
...
-	if targetClusters == nil {
-		return "", fmt.Errorf("target cluster passed is not available to the account/tenant %v", err)
-	}
-	for i := 0; i < len(targetClusters); i++ {
-		if targetClusters[i].GetClusterId() == clusterID {
-			deploymentTargetID = targetClusters[i].GetId()
-			log.Infof("deploymentTargetID %v", deploymentTargetID)
-			log.InfoD("Cluster ID: %v, Name: %v,Status: %v", targetClusters[i].GetClusterId(), targetClusters[i].GetName(), targetClusters[i].GetStatus())
-			targetClusterStatus = targetClusters[i].GetStatus()
+	err = wait.Poll(DefaultRetryInterval, 5*time.Minute, func() (bool, error) {
```
Comment (Madanagopal19) [inline drivers/pds/targetcluster/targetcluster.go:72]: add a constant for defaulttime out

**Review by Madanagopal19 — APPROVED** [review]
LGTM

**Review by aravindranath14 — APPROVED** [review]
LGTM

**Pull request merged**